server/src/Controller/Channel/ChannelController.js

```diff
@@ -0,0 +1,99 @@
+import {
+  create,
+  deleteDoc,
+  findOne,
+  findMany,
+  update,
+} from '../../Repositories/genericRepository';
+import successHandler from '../../libs/routes/successHandler';
+import { channelModel } from '../../Model';
+
+
+const getChannel = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const result = await findOne(channelModel, id);
+    if (!result) {
+      const err = new Error('Channel does not exists');
+      next(err);
+    }
+    res
```

Comment:
Another validation is required like
1. if this id belongs to the user or not, if it is public or private.